src/modules/folder/folder.repository.ts

```diff
@@ -106,15 +107,34 @@ export class SequelizeFolderRepository implements FolderRepository {
     const folders = await this.folderModel.findAll({
       where: { parentId, userId, deleted: deleted ? 1 : 0 },
     });
-    return folders.map((folder) => this._toDomain(folder));
+    return folders.map((folder) => this.toDomain(folder));
+  }
+
+  async findAllByParentId(
+    parentId: FolderAttributes['parentId'],
+    deleted: FolderAttributes['deleted'],
+    page: number = null,
+    perPage: number = null,
+  ): Promise<Array<Folder> | []> {
+    const query: any = {
+      where: { parentId, deleted: deleted ? 1 : 0 },
+      order: [['id', 'ASC']],
+    };
+    const { offset, limit } = Pagination.calculatePagination(page, perPage);
+    if (page && perPage) {
+      query.offset = offset;
+      query.limit = limit;
+    }
+    const folders = await this.folderModel.findAll(query);
+    return folders.map((folder) => this.toDomain(folder));
   }
   async findById(folderId: FolderAttributes['id']): Promise<Folder> {
     const folder = await this.folderModel.findOne({
       where: {
         id: folderId,
       },
     });
-    return this._toDomain(folder);
+    return folder ? this.toDomain(folder) : null;
```

Comment:
According to the type, this should always return a Folder, which seems to be not the case, please reflect on the return type:

`Promise<Folder> -> Promise<Folder | null>`